src/views/clusteroverview/SettingsTab/UserTab/components/ManageSSHKeySection/SSHAuthKeysList/hooks/useSSHAuthKeys.ts

```diff
@@ -56,15 +56,12 @@ const useSSHAuthKeys: UseSSHAuthKeys = () => {
           }),
         );
 
-        filterAuthRows(authRows)
-          .then((filteredRows) => {
-            setAuthKeyRows(filteredRows);
-          })
-          .finally(() => {
-            setLoading(false);
-          });
+        filterAuthRows(authRows).then((filteredRows) => {
+          setAuthKeyRows(filteredRows);
+        });
       }
 
+      setLoading(false);
```

Comment:
this way the loading will be called before the `filterAuthRows` promises get fulfilled or rejected

Reply:
you're right I still need the finally as well as I need it here in case of no keys existing

Reply:
But since it's merged already I'll add it in a follow-up PR